types/json-schema/index.d.ts

```diff
@@ -20,17 +20,23 @@ export type JSONSchema4TypeName = 'string' | 'number' | 'integer' | 'boolean'
  */
 export type JSONSchema4Type = any[] | boolean | number | null | object | string
 
+/**
+ * Meta schema
+ * @see https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-5
+ */
+export type JSONSchema4Version = 'http://json-schema.org/schema#' | 'http://json-schema.org/hyper-schema#'
```

Comment:
From the types side of this, this definition will just be narrowed to `string` and all the other enum values will go away. This looks like the behavior you want so you should just get rid of `JSONSchema4Version` and use `string` in all its use sites. If you want to keep the other urls as documentation you should probably use doc comments to do this, the reasoning being that the compiler will ignore those extra string values but users of your library might want them.

Reply:
Agree with @PranavSenthilnathan